trajopt/src/collision_terms.cpp

```diff
@@ -1620,9 +1652,12 @@ double CollisionCost::value(const sco::DblVec& x)
   {
     const tesseract_collision::ContactResult& res = dist_results[i].get();
     // Contains the contact distance threshold and coefficient for the given link pair
-    const std::array<double, 2>& data =
-        m_calc->getSafetyMarginData()->getPairSafetyMarginData(res.link_names[0], res.link_names[1]);
-    out += sco::pospart(data[0] - dists[i]) * data[1];
+    const auto& collision_config = m_calc->getCollisionConfig();
+    const double dist = collision_config.contact_manager_config.margin_data.getPairCollisionMargin(res.link_names[0],
+                                                                                                   res.link_names[1]);
+    const double coeff =
+        collision_config.collision_coeff_data.getPairCollisionCoeff(res.link_names[0], res.link_names[1]);
+    out += sco::pospart(dist - dists[i]) * coeff;
```

Comment:
Why does this calculation not include the `collision_margin_buffer`?

Reply:
Add a comment to code